src/imports.rs

```diff
@@ -250,13 +250,29 @@ impl<'a> FmtVisitor<'a> {
         self.last_pos = pos_after_last_use_item;
     }
 
-    pub fn format_import(&mut self, vis: &ast::Visibility, vp: &ast::ViewPath, span: Span) {
+    pub fn format_import(
+        &mut self,
+        vis: &ast::Visibility,
+        vp: &ast::ViewPath,
+        span: Span,
+        attrs: &[ast::Attribute],
+    ) {
         let vis = utils::format_visibility(vis);
         // 4 = `use `, 1 = `;`
         let rw = Shape::indented(self.block_indent, self.config)
             .offset_left(vis.len() + 4)
             .and_then(|shape| shape.sub_width(1))
-            .and_then(|shape| vp.rewrite(&self.get_context(), shape));
+            .and_then(|shape| match vp.node {
```

Comment:
This match here is annoying, but I can't pass `attrs` into the `rewrite` impl where this decisions really should be happening.

Would it make sense to add the `attrs` to the `RewriteContext` so that we can handle this case?